### ISSUE TITLE
Get rid of dependency on libglade

### ISSUE THREAD

**Issue by markuspg** [GnoTime/gnotime#4]
Glade is not supported with GTK 3 anymore and should be replaced by a more current mechanism.

**Comment by oskarb** [comment]
As I understand it, glade is supported (and recommended?) for GTK 3.

One approach to explore might be to use glade 3.8 to convert from glade files to gtkbuilder files, as these are supported from GTK+ 2.12. Then continue from there towards GTK3.

**Comment by markuspg** [comment]
Hi @oskarb,

I encountered multiple issues when trying to migrate to GtkBuilder (that's one reason why I decided to move the widgets' definitions into the C codebase).

The main blocker were the widgets of `libgnome(ui)` included into the glade files. They caused many errors on my conversion attempts and prevented the upgrade to GTK 3.

Furthermore I disliked that the glade and GtkBuilder files were often so different, that they could not be compared by my diff tools.

Lastly I find the compiler and the deprecation notices of the GTK codebase immensely useful when upgrading GTK versions. I see the benefit of having a graphical tool to edit UI definitions (self-evidently that makes sense ;-)), so it's a bit difficult how to move forward. Like in the other pull request I would suggest a call to plan how we tackle these issues.

**Comment by oskarb** [comment]
Yes, moving to GTK 3 using GtkBuilder pretty much directly from the current state of the glade files sounds like it would be a mess.

The glade files seems pretty old even for GTK 2 - they appear to be from early GTK 2. I had the idea to first replace use of libgnomeui and deprecated GTK widget, using other GTK 2 widgets instead (which are still available in GTK 3). So a clean GTK 2 version based on GTK 2.24 and without using deprecated widget. I hoped that would then enable moving to GtkBuilder, still on GTK 2. And then on to GTK 3. And that this step-wise process would be manageable.

I browsed a bit in your repo, but it's a lot to look at. :) I did have a look at the converting-glade-to-code changes that you did. It is quite a lot of code and I worry that it will be tedious to maintain. Have you tried some approach like the above? But I don't know what the long-term best strategy would be... GTK 4 still uses GtkBuilder - but Glade will not be updated beyound GTK 3. Though there are attempts at other tools.

I created issues #12 and #13 while I tried to understand what the problematic widgets were, with some thoughts.
